### PR TITLE
Doc/add sphinx doc

### PR DESCRIPTION
**Pull request by raphaelventura**
This PR adds the ability to build Read the docs-style documentation using `sphinx`.

Code docstrings were updated accordingly and notes were added inside the README and data_science_workflow files.

I did not add a git hook for automatic generation yet since the `build/` folder weighs as much as 15MB (!), and I don't think it should be pushed to github. Maybe this could be done as gh pages outside of this repo. What do you think @vviers ?

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import os
+import sys
+
+sys.path.insert(0, os.path.abspath("../predictsignauxfaibles/"))
+
+
+# -- Project information -----------------------------------------------------
+
+project = "predictsignauxfaibles"
+copyright = "2021, Simon Lebastard, Raphaël Ventura, Vincent Viers"
```
Comment (vviers) [inline docs/source/conf.py]: A verifier mais je ne sais pas si le projet est copyrighté ? Vu qu'il est open-source ?

Comment (vviers) [reply docs/source/conf.py]: Dans tous les cas je dirais `copyright = signaux-faibles`

**Review by vviers — DISMISSED** [review]
LGTM :)
Thank you very much for this great work 💯

File: docs/source/conf.py (hunk 1 above)
Comment (raphaelventura) [reply docs/source/conf.py]: Oui, on peut effectivement mettre «The signaux-faibles contributors» pour plus de généralité.

**Pull request merged**